### PR TITLE
Add MPI timed test

### LINKED ISSUE #2190: [TASK] Add ability to specify mpi parameters
--------
Issue Description
--------

**Is your feature request related to a problem? Please describe.**
Sometimes it is useful to be able to specify parameters to the mpiexec used by RAVEN.

**Describe the solution you'd like**
Add something to the xml that allows this.

**Describe alternatives you've considered**
Dump it in some other part of the command so it ends up as a parameter, but this can make the input somewhat confusing.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by joshua-cogliati-inl**
--------
Pull Request Description
--------
##### What issue does this change request address? 
Closes #2190

##### What are the significant changes in functionality due to this change request?
This adds a test for mpi running that is compute bound, to check if slowdowns have occured.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: tests/cluster_tests/tests [hunk 1]
```diff
@@ -26,6 +26,18 @@
...
  [../]
+ [./test_mpiqsub_long]
+   input='test_mpiqsub_long.xml cluster_runinfo_legacy.xml'
+   run_types = 'qsub'
+   output = 'FirstMQL/FirstMQRun/1/out~simple_gp_test.csv FirstMQL/FirstMQRun/6/out~simple_gp_test.csv'
+   csv = 'FirstMQL/samples.csv'
+   output_wait_time = 60
+   #Note that the output includes the time the test is running
+   # which is why the relative error is very high
+   rel_err = 0.1
```
Comment (joshua-cogliati-inl) [inline tests/cluster_tests/tests:37]: Possibly should bump this to 0.2

File: tests/cluster_tests/test_mpiqsub_long.xml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    <author>cogljj</author>
+    <created>2015-02-26</created>
+    <classesTested>MPISimulationMode</classesTested>
+    <description>
+      This tests running with mpi and a qsub.
+    </description>
+    <revisions>
+      <revision author="alfoa" date="2018-10-01">This is a requirement test now.</revision>
+    </revisions>
+    <requirements>R-IS-7</requirements>
```
Comment (wangcj05) [inline tests/cluster_tests/test_mpiqsub_long.xml:14]: Please update the test info

File: tests/cluster_tests/compute_test.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python
+# Copyright 2023 Battelle Energy Alliance, LLC
+#
```
Comment (wangcj05) [inline tests/cluster_tests/compute_test.py:3]: It seems to me this is inconsistent with RAVEN, I mean change 2023 to 2017 here?

File: doc/user_manual/runInfo.tex [hunk 4]
```diff
@@ -327,6 +327,7 @@ \subsection{RunInfo: Input of Queue Modes}
       in~\cite{MPI}.
       Mode ``MPI''  can either generate a \texttt{qsub} command or can execute
       on selected nodes.
+      Parameters to be given to the mpi command can be specified with the \xmlNode{MPIParam} node.
```
Comment (wangcj05) [inline doc/user_manual/runInfo.tex:330]: Could you provide more descriptions here? In addition, could you add this option to a test?

**Review by wangcj05 — CHANGES_REQUESTED** [review]
@joshua-cogliati-inl Some minor comments for you to consider.

File: tests/cluster_tests/compute_test.py (hunk 3 above)
Comment (joshua-cogliati-inl) [reply tests/cluster_tests/compute_test.py]: Hm, changed, but I am not sure it is needed.

File: tests/cluster_tests/test_mpiqsub_long.xml (hunk 2 above)
Comment (joshua-cogliati-inl) [reply tests/cluster_tests/test_mpiqsub_long.xml]: Done.

**Review by wangcj05 — APPROVED** [review]
changes look good.

**Comment by wangcj05** [comment]
@joshua-cogliati-inl Can you create an issue for this PR?

**Comment by wangcj05** [comment]
PR looks good, and checklist is satisfied.

**Pull request closed without merging**